interfaces/acados_matlab_octave/AcadosOcp.m

```diff
@@ -257,6 +257,13 @@ function make_consistent(self)
                 dims.nbxe_0 = 0;
             end
 
+            % check if has_x0 is true
+            % if true ineqaulities are set to equalities
+            if dims.nx == dims.nbx_0 && all(sort(reshape(constraints.idxbx_0, 1, [])) == 0:(dims.nx-1)) && all(constraints.lbx_0 == constraints.ubx_0)
+                constraints.has_x0 = true;
+                dims.nbxe_0 = dims.nx;
```

Comment:
If we add this, it is not possible to have an x0 constraint without marking it as an equality constraint. Right?
This concerns me a bit, since I think this is sometimes desired.
Why did you add this?
Could we just have the check in the transription from the legacy interface maybe?